### PR TITLE
python: don't run mkdirp in `setup_dependent_package`

### PR DESCRIPTION
**Pull request by haampie**
Extracted from #40773

`setup_dependent_package` is not a build phase, it should just set
globals for a package.

It's called during setup of runtime environment of packages, and there
have been reports of it actually failing on Slack.

### CODE REVIEW

**Review by adamjstewart — DISMISSED** [review]
I'm personally fine with this if CI passes, but would like at least one more review. Can you track down who added this originally and why they added it?

File: var/spack/repos/builtin/packages/py-tensorflow/package.py [hunk 1]
```diff
@@ -856,6 +856,7 @@ def post_configure_fixes(self):
         filter_file("build:opt --host_copt=-march=native", "", ".tf_configure.bazelrc")
 
     def build(self, spec, prefix):
+        mkdirp(python_platlib)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-tensorflow/package.py:860]: Can you add a comment on why this is necessary?

**Comment by haampie** [comment]
> Can you track down who added this originally and why they added it?

Added in PR #18, cause it was considered "simpler", or just to demo a new feature.

https://github.com/spack/spack/pull/18/commits/bcccf020204a556e382c0af2897ad9126bb24984

I don't think much thought went into that at that point.

**Review by alalazo — APPROVED** [review]
:+1: If Gitlab passes

**Comment by haampie** [comment]
Gitlab won't pass cause of unrelated known failures on develop, so needs a force merge.

**Pull request merged**